### PR TITLE
Finished fourth assignment

### PR DESCRIPTION
**Pull request by MrPranklin**
- [x] Do all the steps in the chapter by yourself.
- [x] In the comments creation form, on the posts show page, add a captcha.
- [x] The captcha should consist of a simple equation with the form x + y - z = n, where n is randomly chosen between 100 and 999.
- [x] All fields of the captcha should be sent to the server as a sub-hash named captcha
- [x] The server chooses the value of n and stores it in a hidden field in the form
- [x] All parameters should be positive integers different from zero
- [x] If the x + y - z aren't equal to n or if any parameter doesn't conform to the specified rules, redirect the user back, using redirect_back(fallback_location: root_path) and display a validation error using a flash notice

### CODE REVIEW

**Comment by Xenosb** [comment]
Please add @Narayanan170 as a reviewer

**Comment by MrPranklin** [comment]
> Please add @Narayanan170 as a reviewer

He still hasn't accepted collaboration invite after 5 days.

**Comment by Narayanan170** [comment]
Will accept after I come back from vacation.

Br,

F

**Comment by Xenosb** [comment]
Ping Suste

File: app/controllers/polynomials_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  def solve(a, b, c, d)
+    require 'cmath'
+
+    a = a.to_r
+    b = b.to_r
+    c = (c - d).to_r
+
+    first = -1 * b
+    second = CMath.sqrt(b**2 - 4 * a * c) / (2 * a)
```
Comment (Narayanan170) [inline app/controllers/polynomials_controller.rb]: Why use "cmath" when you can call `Math.sqrt` which is already loaded in.

Comment (MrPranklin) [reply app/controllers/polynomials_controller.rb]: Math doesn't work with complex roots since it only accepts positive numbers.

**Comment by Narayanan170** [comment]
> Math (without the C) doesn't work with negative input, therefore it does not work for complex roots.

You're right, didn't understand the problem you were trying to solve, sorry.

**Comment by Narayanan170** [comment]
> When I replaced form_tag with simple_form_for in the polynomials new view i got no post method errors when using resources instead of manually written get/post. I added `post 'polynomials/new', to: 'polynomials#create'`

The problem is not with the routes, as create *should* be pointing to /polynomials, the problem is the path of the simple form.

**Comment by Narayanan170** [comment]
Further explanation:
Rails is wonderful, but sometimes it hides things away from you that should not be hidden.
https://guides.rubyonrails.org/form_helpers.html#binding-a-form-to-an-object
See, you are trying to bind the form to the object :operands, which is nothing but a hash. Rails takes a few assumptions here, and then binds the form to the wrong URL.
Since :operands is not a rails model, it binds the :url option to the current path (/polynomials/new). The right way of doing that is just passing the url to bind that form to. 
So, instead of doing this:
`<%= simple_form_for :operands do |f| %>`
You can add the url of the create action for polynomials, so it would point to the right controller/action
`<%= simple_form_for :operands, url: polynomials_path do |f| %>`
Then you can leave the `resources :polynomials, only: [:show, :new, :create]

**Pull request merged**